server/src/main/java/org/opensearch/cluster/metadata/MetadataCreateIndexService.java

```diff
@@ -1497,10 +1497,16 @@ public static void validateTranslogRetentionSettings(Settings indexSettings) {
      * @param clusterSettings cluster setting
      */
     public static void validateRefreshIntervalSettings(Settings requestSettings, ClusterSettings clusterSettings) {
-        if (IndexSettings.INDEX_REFRESH_INTERVAL_SETTING.exists(requestSettings) == false) {
+        if (IndexSettings.INDEX_REFRESH_INTERVAL_SETTING.exists(requestSettings) == false
+            || requestSettings.get(IndexSettings.INDEX_REFRESH_INTERVAL_SETTING.getKey()) == null) {
             return;
         }
         TimeValue requestRefreshInterval = IndexSettings.INDEX_REFRESH_INTERVAL_SETTING.get(requestSettings);
+        // If the refresh interval supplied is -1, we allow the index to be created because -1 means no periodic refresh.
+        if (requestRefreshInterval.millis() < 0) {
+            assert requestRefreshInterval.millis() == -1;
```

Comment:
Nitpick, but I'd remove this. The setting itself defines the legal range and unless you need specific logic built around a constraint related to the `-1` value then its probably better to rely on a single place to enforce min/max.

Reply:
I had added this for just verifying that the only legal value if refresh interval is less than 0 is -1. I was anyways going to remove this assertion and have a stricter check of -1 value.